core/store/src/trie/shard_tries.rs

```diff
@@ -292,8 +300,12 @@ impl ShardTries {
         shard_uid: ShardUId,
         store_update: &mut StoreUpdate,
     ) {
+        // `itoa` is much faster for printing shard_id to a string than trivial alternatives.
+        let mut buffer = itoa::Buffer::new();
+        let shard_id = buffer.format(shard_uid.shard_id);
```

Comment:
To not repeate the same comment three times, extract `get_shard_gauge(metric, shard_id)` helper?